getting-started/cli/recipes.md

```diff
@@ -0,0 +1,152 @@
+$ open http://localhost:3000
+200  | GET  "/"  | 294.0µs
+Params:
+```
+
+# Amber Generate Command
+
+The generators in amber are a great way to get an application up and running quickly.
+In addition, they help keep code consistent and following convention.  See the
+**Generate** option of the **Command Line Tool** for details about the generate command.
+
+Once you have generated an application with a recipe, the same recipe will be used
+by the generate command when you generate controllers, models or scaffold controllers,
+models and views with the scaffold generator.
+
+# Custom Recipe
+
+One of the recipes available from the recipe repository may almost suit your requirements,
+but not quite.  The recipes are just a set of files that are templates for various
+parts of an application.  You can download and extract a recipe and modify the templates
+to better suit your requirements (and then contribute your modified recipe to the repository so
+everyone can benefit from your changes).
+
+Download the recipe and extract it to a folder and then give the path to the folder as
+the recipe command line argument.  For example;
+
+1. `wget https://raw.githubusercontent.com/AmberRecipes/recipes/master/damianham/modular.zip`
+1. `mkdir ~/mymodular`
+1. `unzip -d ~/mymodular modular.zip`
+1. `# modify the recipe in ~/mymodular some way`
+1. `amber new microsecond-blog -r ~/mymodular`
+1. `cd microsecond-blog`
+1. `shards install`
+1. `amber run &` or `amber watch &` (also can execute these in foreground of separate terminal)
+1. `open http://localhost:3000`
+
+It is important to give the full path to the recipe folder as the recipe command line
+argument to ensure that the recipe can be found when generating subsequent artifacts.
+
+# Custom Repository
+
+You can specify the location of a recipe repository in the `.amber.yml` file so that you can use
+recipes from an independent source.  Add the **recipe_source** item to the '.amber.yml' file
+with the URL of the website or website folder that contains the recipe zip files to use the recipes
+from the given URL. E.g
+
+```
+type: app
+database: sqlite
+language: slang
+model: granite
+recipe: myrecipe
+recipe_source: http://my.example.org/recipes
+```
+
+The scaffold generator would then use a recipe located at
+`http://my.example.org/recipes/myrecipe.zip`
+
+
+## Clear Recipe Cache
+
+If you set a recipe source URL you should also change the name of the recipe and/or
+you should clear the recipe cache otherwise a recipe with the same name in the
+recipe cache will be used rather than a recipe from the given recipe source.
+
+The recipe cache is located in the current working directory and is created any time
+you specify a recipe name that is downloaded from the repository.  Since you would
+usually specify a recipe when creating a new application the recipe cache will be
+located in the parent folder of the new application in a folder called `.amber_recipe_cache`.
```

Comment:
Perhaps this isn't the right pull request to be having this conversation, but I'm not sure I like just dropping the amber recipe cache in random places on computers.

Reply:
noted in PR #710